packages/fuels-contract/src/contract.rs

```diff
@@ -694,6 +735,19 @@ impl MultiContractCallHandler {
                         .take(1)
                         .for_each(|call| call.append_variable_outputs(1));
                 }
+
+                Err(Error::RevertTransactionError(_, ref receipts)) => {
+                    if let Some(receipt) = ContractCall::find_contract_not_in_inputs(receipts) {
+                        let contract_id = Bech32ContractId::from(*receipt.contract_id().unwrap());
+                        self.contract_calls
+                            .iter_mut()
+                            .take(1)
```

Comment:
Wait, what's the purpose of this `take(1)` in this context?

Reply:
There are more `ContractCalls` in `pub contract_calls: Vec<ContractCall`. In this way, I put the `contract_id` only in the `ContractCall` where it should be, that is, in the one that calls the missing `contract_id`.